engine/gamesys/src/gamesys/components/comp_model.cpp

```diff
@@ -1196,12 +1386,44 @@ namespace dmGameSystem
                 }
             }
 
-            FillTextures(&ro, component, material_index);
+            int32_t first_free_index = FillTextures(&ro, component, material_index);
+
+            HComponentRenderConstants constants = component->m_RenderConstants;
+
+            if (component->m_RigInstance && render_item->m_Buffers->m_RigModelVertexFormat == RIG_MODEL_VERTEX_FORMAT_SKINNED)
+            {
+                if (first_free_index >= 0)
+                {
+                    dmRender::SetMaterialSampler(ro.m_Material, NAME_POSE_MATRIX_CACHE, first_free_index,
+                        dmGraphics::TEXTURE_WRAP_CLAMP_TO_EDGE, dmGraphics::TEXTURE_WRAP_CLAMP_TO_EDGE,
+                        dmGraphics::TEXTURE_FILTER_NEAREST, dmGraphics::TEXTURE_FILTER_NEAREST, 0.0f);
+                    ro.m_Textures[first_free_index] = world->m_SkinnedAnimationData.m_BindPoseCacheTexture;
+                }
+                else
+                {
+                    dmLogOnceError("Unable to bind bone matrix cache texture for component '%s', no free texture slot available.",
+                        dmHashReverseSafe64(dmGameObject::GetIdentifier(component->m_Instance)));
+                }
+
+                constants = constants ? constants : world->m_SkinnedAnimationData.m_AnimationRenderConstants;
+
+                // *4 = 4 vectors per matrix (RGBA)
+                uint32_t cache_offset = 4 * dmRig::GetPoseMatrixCacheDataOffset(world->m_RigContext, component->m_RigInstance);
```

Comment:
I'm used to only storing 3*4 = 12 floats (as the last column is (0,0,0,1)), to save bandwidth. (doing a transpose to make it into 3 vectors)
Is that something we want to do or perhaps it's not valid these days?
(may still be a mvp2 thing, but not sure when we'll get to that again)

Reply:
I couldn't get it to work so I'm pausing it for now in favor of merging this PR. I added a note about it in the code for future reference.